### ISSUE TITLE
Potentially broken use of JSON::decode_json

### ISSUE THREAD

**Issue by TiMBuS** [perl-carton/carton#214]
Hello!
This code here: https://github.com/perl-carton/carton/blob/master/lib/Carton/Util.pm#L22 is going to fail if the JSON::XS module is `use`d or otherwise imported with function prototypes.

To quickly demonstrate:
```
perl -MJSON::XS -e '@zuh = "{}"; JSON::XS::decode_json(@zuh)'
malformed JSON string, neither tag, array, object, number, string or atom, at character offset 0 (before "(end of string)") at -e line 1.
```

To fix it you can either pass single scalar argument to the function, or call the function without the prototype enabled:
```perl
JSON::decode_json($_[0]);
# or
&JSON::decode_json(@_);
```
I'd recommend the latter.

**Comment by miyagawa** [comment]
Why the latter? It seems to me explicitly passing argument is better.

**Comment by TiMBuS** [comment]
Well since it's mostly just a shim, I figured you maybe wanted to pass in all of the arguments and let the underlying lib deal with it.

**Comment by putaotao** [comment]
This will let the params confusion. Explicit params better. 


发自网易邮箱 iPhone 版
在2016年09月26日 09:07，Jarrod Funnell 写道:

Well since it's mostly just a shim, I figured you maybe wanted to pass in all of the arguments and let the underlying lib deal with it.

**Comment by miyagawa** [comment]
Fixed with f13faa8
